### PR TITLE
V3 fixes

### PR DESCRIPTION
**Pull request by asdcdow**
Some more changes to support Swagger v3.  Contains some documentation changes and some fixes to support the format for parameters used by v3

### CODE REVIEW

File: core/src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala [hunk 1]
```diff
@@ -166,15 +166,27 @@ final case class SwaggerSpecGenerator(
...
-    (__ \ 'required).write[Boolean] ~
-    (__ \ 'default).writeNullable[JsValue] ~
-    (__ \ 'example).writeNullable[JsValue] ~
-    (__ \ "schema").writeNullable[String](refWrite) ~
-    (__ \ "items").writeNullable[SwaggerParameter](propWrites) ~
-    (__ \ "enum").writeNullable[Seq[String]])(unlift(GenSwaggerParameter.unapply))
+    if (swaggerV3) {
+      (__ \ 'name).write[String] ~
+      (__ \ "schema").writeNullable[String](refWrite) ~
+      (__ \ "schema" \ 'type).writeNullable[String] ~
```
[the remark this answers is not in the document]
Comment (asdcdow) [reply core/src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala]: Seems like a good idea.  I'll fix that.

**Comment by asdcdow** [comment]
Is that better?

File: core/src/main/scala/com/iheart/playSwagger/Domain.scala [hunk 2]
```diff
@@ -22,12 +22,12 @@ object Domain {
 
   final case class GenSwaggerParameter(
     name:          String,
+    referenceType: Option[String]           = None,
     `type`:        Option[String]           = None,
     format:        Option[String]           = None,
     required:      Boolean                  = true,
     default:       Option[JsValue]          = None,
     example:       Option[JsValue]          = None,
-    referenceType: Option[String]           = None,
```
Comment (kailuowang) [inline core/src/main/scala/com/iheart/playSwagger/Domain.scala]: is this change necessary?

File: core/src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala [hunk 3]
```diff
@@ -173,16 +173,31 @@ final case class SwaggerSpecGenerator(
...
-    (__ \ "schema").writeNullable[String](refWrite) ~
-    (__ \ "items").writeNullable[SwaggerParameter](propWrites) ~
-    (__ \ "enum").writeNullable[Seq[String]])(unlift(GenSwaggerParameter.unapply))
+  private lazy val genParamWrites: OWrites[GenSwaggerParameter] = {
+    val commonFields = (__ \ 'name).write[String] ~
+        (__ \ "schema").writeNullable[String](refWrite)
+
+
+    (if (swaggerV3) {
+      commonFields ~
```
Comment (kailuowang) [inline core/src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala]: I am thinking maybe you could have a 
`val under = if (swaggerV3) __ \ "schema" else __`
then you can just do 
```scala
(under \ 'type).writeNullable[String] ~
(under \ 'format).writeNullable[String] ~
(under \ 'required).write[Boolean] ~
(under \ 'default).writeNullable[JsValue] ~
(under \ 'example).writeNullable[JsValue] ~
(under \ "items").writeNullable[SwaggerParameter](propWrites) ~
(under \ "enum").writeNullable[Seq[String]]
```
right?

File: core/src/main/scala/com/iheart/playSwagger/Domain.scala (hunk 2 above)
Comment (asdcdow) [reply core/src/main/scala/com/iheart/playSwagger/Domain.scala]: I updated the order of the fields in `genParamWrites` so I had to update the order of the fields here so that `GenSwaggerParameter.unapply` would have the correct field order.

File: core/src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala (hunk 3 above)
Comment (asdcdow) [reply core/src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala]: That is a good idea.  I hadn't considered that.  I'll change it.

**Comment by asdcdow** [comment]
Better?

File: core/src/main/scala/com/iheart/playSwagger/Domain.scala (hunk 2 above)
Comment (kailuowang) [reply core/src/main/scala/com/iheart/playSwagger/Domain.scala]: i see.

**Comment by kailuowang** [comment]
thanks very much. Let me know when you think this is ready for a release. (will need to merge this to the play 2.6 branch)

**Comment by asdcdow** [comment]
I noticed 2 more problems, `required` is in the wrong place for params.  And customer types for params are not placed properly. I will have a pull request for those later today and then it should be ready.
